File: java/src/test/java/ObserverTest.java
import observer.Messenger;
import observer.sender.AppAlarmSender;
import observer.sender.MailSender;
import org.junit.jupiter.api.Test;

public class ObserverTest {
    @Test
    void observerRun() {
       var messenger = new Messenger();
       var mailSender = new MailSender();
       var appAlarmSender = new AppAlarmSender();

       messenger.subscribe(mailSender);
       messenger.subscribe(appAlarmSender);

       messenger.publish("Show me the Money!!");

       messenger.unsubscribe(appAlarmSender);

       messenger.publish("Give me the Money!");
    }
}

File: src/test/kotlin/ObserverTest.kt
import io.kotest.core.spec.style.FunSpec
import observer.Messenger
import observer.sender.AppAlarmSender
import observer.sender.MailSender

class ObserverTest : FunSpec({
    test("Test Observer") {
        val messenger = Messenger()
        val mailSender = MailSender()
        val appAlarmSender = AppAlarmSender()

        messenger.subscribe(mailSender)
        messenger.subscribe(appAlarmSender)

        messenger.publish("Show me the Money!")

        messenger.unsubscribe(appAlarmSender)

        messenger.publish("Give me the Money!")
    }
})